### PR TITLE
fix(jsx): replace jsx usage with createComponent and useElement

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

Replace the deprecated `jsx` function with `createComponent` and `useElement` in every component using it.

> 💡Tip to review: compare the [dev branch storybook](https://620e870c28c4ae003a705d42-jrkxxaefnm.chromatic.com/?path=/story/admin-ui-core-core--frameworkd-agnostic) with the storybook generated in this PR to check if they have the same output.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/components/Modal/components/ModalFooter.tsx [hunk 1]
```diff
@@ -17,51 +17,62 @@ import { useModalContext } from './ModalContext'
...
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      borderTop: '$neutral',
+      position: 'sticky',
+      bottom: 0,
+      left: 0,
+      right: 0,
+      bg: '$primary',
+      padding: 6,
+      flexDirection: ['column-reverse', 'column-reverse', 'row'],
```
Comment (Lenderson) [inline packages/admin-ui/src/components/Modal/components/ModalFooter.tsx:37]: is this for the "responsiveness array"?

**Review by matheusps — APPROVED** [review]

File: packages/admin-ui/src/components/Modal/components/ModalFooter.tsx (hunk 1 above)
Comment (matheusps) [reply packages/admin-ui/src/components/Modal/components/ModalFooter.tsx:37]: I guess you should change it for the responsive aliases

Comment (matheusps) [reply packages/admin-ui/src/components/Modal/components/ModalFooter.tsx:37]: And we have a bug about it: https://vtex.slack.com/archives/C01DVTFA4VA/p1658162740473519.

Comment (lucasaarcoverde) [reply packages/admin-ui/src/components/Modal/components/ModalFooter.tsx:37]: yes! I've probably missed it :squirrel: 
I'll change, good catch!

**Review by Lenderson — APPROVED** [review]

**Pull request merged**